src/repl/scala/tools/nsc/interpreter/Imports.scala

```diff
@@ -127,7 +127,8 @@ trait Imports {
           case rh :: rest if !keepHandler(rh.handler) => select(rest, wanted)
           case rh :: rest                             =>
             import rh.handler._
-            val newWanted = wanted ++ referencedNames -- definedNames -- importedNames
+            val augment = rh match { case ReqAndHandler(_, _: ImportHandler) => referencedNames ; case _ => Nil }
```

Comment:
This part is unclear to me... the intention here is to add referenced names *only* from other imports? Say, we discovered we need to add `import foo.bar`, because `bar` is wanted`, but we also need to import `foo`, in case it comes from a previous request?